boost-maven/boost-runtimes/runtime-liberty/runtime-liberty-common/src/main/java/boost/runtimes/liberty/LibertyRuntime.java

```diff
@@ -37,54 +37,51 @@
 import boost.runtimes.openliberty.boosters.LibertyBoosterI;
 
-public class LibertyRuntime implements RuntimeI {
-    private final List<AbstractBoosterConfig> boosterConfigs;
-    private final ExecutionEnvironment env;
-    private final MavenProject project;
-    private final Plugin mavenDepPlugin;
-
+public abstract class LibertyRuntime implements RuntimeI {
     private final String serverName = "BoostServer";
-    private final String projectBuildDir;
-    private final String libertyServerPath;
+    private final String serversDir = "/liberty/wlp/usr/servers/";
+
+    private final String runtimeGroupId;
+    private final String runtimeArtifactId;
+    private final String runtimeVersion;
 
-    private final String runtimeGroupId = "io.openliberty";
-    private final String runtimeArtifactId = "openliberty-runtime";
-    private final String runtimeVersion = "19.0.0.3";
+    private String libertyServerPath;
+
+    private MavenProject project;
+    private ExecutionEnvironment env;
 
     private String libertyMavenPluginGroupId = "net.wasdev.wlp.maven.plugins";
     private String libertyMavenPluginArtifactId = "liberty-maven-plugin";
     private String libertyMavenPluginVersion = "2.6.3";
 
-    public LibertyRuntime() {
-        this.boosterConfigs = null;
-        this.env = null;
-        this.project = null;
-        this.projectBuildDir = null;
-        this.libertyServerPath = null;
-        this.mavenDepPlugin = null;
-    }
+    private String mavenDependencyPluginGroupId = "org.apache.maven.plugins";
+    private String mavenDependencyPluginArtifactId = "maven-dependency-plugin";
 
-    public LibertyRuntime(RuntimeParams runtimeParams) {
-        this.boosterConfigs = runtimeParams.getBoosterConfigs();
-        this.env = runtimeParams.getEnv();
-        this.project = runtimeParams.getProject();
-        this.projectBuildDir = project.getBuild().getDirectory();
-        this.libertyServerPath = projectBuildDir + "/liberty/wlp/usr/servers/" + serverName;
-        this.mavenDepPlugin = runtimeParams.getMavenDepPlugin();
+    public LibertyRuntime(String runtimeGroupId, String runtimeArtifactId, String runtimeVersion) {
+        this.runtimeGroupId = runtimeGroupId;
+        this.runtimeArtifactId = runtimeArtifactId;
+        this.runtimeVersion = runtimeVersion;
     }
 
     private Plugin getPlugin() throws MojoExecutionException {
         return plugin(groupId(libertyMavenPluginGroupId), artifactId(libertyMavenPluginArtifactId),
                 version(libertyMavenPluginVersion));
     }
 
+    public ExecutionEnvironment getExecutionEnvironment() {
+        return env;
+    }
+
     @Override
-    public void doPackage() throws BoostException {
+    public void doPackage(List<AbstractBoosterConfig> boosterConfigs, Object mavenProject, Object pluginTask) throws BoostException {
+        project = (MavenProject)mavenProject;
+        env = ((AbstractMojo)pluginTask).getExecutionEnvironment();
```

Comment:
I think all the casting suggests that we should have two distinct _interfaces_  one from boost-maven and one from boost-gradle, and let the runtime impls decide if and how they want to factor the common pieces.   The interface isn't doing a good job at showing what needs to be implemented.